File: BlogApi/Models/BlogPost.cs
using System.ComponentModel.DataAnnotations;

namespace BlogApi.Models;

public class BlogPost
{
    public int Id { get; set; }

    [Required]
    public string Title { get; set; } = "";

    public string Content { get; set; } = "";

    public string Author { get; set; } = "";

    public DateTime CreatedDate { get; set; }

    public bool IsPublished { get; set; }
}

File: BlogFrontend/src/app/models/blog-post.ts
export interface BlogPost {
  id: number;
  title: string;
  content: string;
  author: string;
  createdDate: string;
  isPublished: boolean;
}

export interface CreateBlogPostDto {
  title: string;
  content: string;
  author: string;
  isPublished: boolean;
}

export interface UpdateBlogPostDto {
  title: string;
  content: string;
  author: string;
  isPublished: boolean;
}

export interface PaginatedResponse<T> {
  page: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
  items: T[];
}

